core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -636,6 +640,27 @@ private SqlCharStringLiteral creteDateTimeFormatSqlCharLiteral(String format) {
       String standardDateFormat, Map<SqlDateTimeFormat, String> dateTimeFormatMap) {
     return super.getDateTimeFormatString(standardDateFormat, dateTimeFormatMap);
   }
+
+  private void unparsePlusMinus(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    switch (call.getOperator().getName()) {
+    case "TIMESTAMP_ADD":
+    case "TIMESTAMP_SUB":
+      call.operand(0).unparse(writer, leftPrec, rightPrec);
+      checkSign(writer, call);
+      call.operand(1).unparse(writer, leftPrec, rightPrec);
+      break;
+    default:
+      super.unparseCall(writer, call, leftPrec, rightPrec);
+    }
+  }
+
+  private void checkSign(SqlWriter writer, SqlCall call) {
+    if (call.getKind() == SqlKind.PLUS) {
```

Comment:
switch known value to left side of comparison

Reply:
done